assets/src/edit-story/app/media/utils/getVideoDuration.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Returns the duration of a given video url
+ *
+ * @param {string} src Video src URL.
+ * @return {Promise<string>} The extracted image in base64-encoded format.
+ */
+function getVideoDuration(src) {
+  const video = document.createElement('video');
+  video.muted = true;
+  video.crossOrigin = 'anonymous';
+  // Since  we want to get the actual frames, we need to make sure to preload the whole video
+  // and not just metadata.
+  // See https://github.com/google/web-stories-wp/issues/2922.
+  video.preload = 'auto';
+
+  return new Promise((resolve, reject) => {
+    video.addEventListener('error', reject);
+
+    video.addEventListener('loadedmetadata', () => {
```

Comment:
Should we create a named function and remove the event listener when it `resolve`s?

Reply:
I followed the code of this file https://github.com/rtCamp/web-stories-wp/blob/main/assets/src/edit-story/app/media/utils/getFirstFrameOfVideo.js. I believe we don't need any remove listener because we are attaching the listener to the specific `video` that created and later destroyed by garbage collector within this function `getVideoDuration`.